File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.cpp
class Solution {
public:
    int minGroups(vector<vector<int>>& intervals) {
        
        int n = intervals.size();

        sort(intervals.begin(), intervals.end());

        priority_queue<int, vector<int>, greater<int>> minEnd;

        for(int i=0; i<n; i++)
        {
            int start = intervals[i][0];
            int end = intervals[i][1];

            if(!minEnd.empty() && start > minEnd.top())
            {
                minEnd.pop();
            }
            minEnd.push(end);

        }
        return minEnd.size();
    }
};

File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.java
class Solution {
    public int minGroups(int[][] intervals) {
        
        int n = intervals.length;
        
        Arrays.sort(intervals, Comparator.comparingInt(a -> a[0]));
        
        PriorityQueue<Integer> minEnd = new PriorityQueue<>();


        for(int i=0; i<n; i++)
        {
            
            int start = intervals[i][0];
            int end = intervals[i][1];

            if(!minEnd.isEmpty() && minEnd.peek() < start)
            {
                minEnd.poll();
            }
            minEnd.add(end);
           
        }
        return minEnd.size();
    }
}